velox/exec/SortBuffer.cpp

```diff
@@ -117,20 +119,30 @@ void SortBuffer::noMoreInput() {
     // the rows.
     sortedRows_.resize(numInputRows_);
     RowContainerIterator iter;
-    data_->listRows(&iter, numInputRows_, sortedRows_.data());
-    std::sort(
-        sortedRows_.begin(),
-        sortedRows_.end(),
-        [this](const char* leftRow, const char* rightRow) {
-          for (vector_size_t index = 0; index < sortCompareFlags_.size();
-               ++index) {
-            if (auto result = data_->compare(
-                    leftRow, rightRow, index, sortCompareFlags_[index])) {
-              return result < 0;
+    if (prefixSortConfig_.has_value()) {
+      auto prefixSort = PrefixSort(
+          data_.get(),
+          sortCompareFlags_,
+          numInputRows_,
+          prefixSortConfig_.value());
+      prefixSort.preparePrefix();
+      prefixSort.sort(sortedRows_);
+    } else {
+      data_->listRows(&iter, numInputRows_, sortedRows_.data());
+      std::sort(
+          sortedRows_.begin(),
+          sortedRows_.end(),
+          [this](const char* leftRow, const char* rightRow) {
+            for (vector_size_t index = 0; index < sortCompareFlags_.size();
```

Comment:
For a simple case sorting by a few integers, we can optimize this comparison to use memcmp, no? I wonder how would this optimization compare with using PrefixSort. Specifically, I'm curious whether it will give us the most benefits with PrefixSort adding just a little bit of perf boost on top.

Reply:
Oh, this is the code velox use now  see the if body that`s PrefixSort impl
`
if (prefixSortConfig_.has_value()) 
`

Reply:
That's right and I assume this is very inefficient. I wonder what happens if we just optimize this comparator. I expect it will be speed things up quite a bit.